src/main/scala/org/ergoplatform/nodeView/ErgoNodeViewHolder.scala

```diff
@@ -612,12 +626,54 @@ object ErgoNodeViewHolder {
 
     case class EliminateTransactions(ids: Seq[scorex.util.ModifierId])
 
+    case object IsChainHealthy
+    sealed trait HealthCheckResult
+    case object ChainIsHealthy extends HealthCheckResult
+    case class ChainIsStuck(reason: String) extends HealthCheckResult
   }
 
+  case class BlockAppliedTransactions(txs: Seq[scorex.util.ModifierId]) extends NodeViewHolderEvent
+
   case class DownloadRequest(modifierTypeId: ModifierTypeId,
                              modifierId: scorex.util.ModifierId) extends NodeViewHolderEvent
 
   case class CurrentView[State](history: ErgoHistory, state: State, vault: ErgoWallet, pool: ErgoMemPool)
+
+  /**
+    * Checks whether chain got stuck by comparing timestamp of bestFullBlock or last time a modifier was applied to history.
+    * @param progress metadata of last chain update
+    * @return ChainIsHealthy if chain is healthy and ChainIsStuck(error) with details if it got stuck
+    */
+  def checkChainIsHealthy(
+      progress: ChainProgress,
+      history: ErgoHistory,
+      context: ErgoStateContext,
+      settings: ErgoSettings): HealthCheckResult = {
+    val ChainProgress(lastMod, headersHeight, blockHeight, lastUpdate) = progress
+    val chainUpdateDelay = System.currentTimeMillis() - lastUpdate
+    val acceptableChainUpdateDelay = settings.nodeSettings.acceptableChainUpdateDelay
+    def chainUpdateDelayed = chainUpdateDelay > acceptableChainUpdateDelay.toMillis
+    def blockUpdateDelayed =
+      history.bestFullBlockOpt
+        .map(b => System.currentTimeMillis() - b.header.timestamp)
+        .exists(blockUpdateDelay => blockUpdateDelay > acceptableChainUpdateDelay.toMillis)
+
+    def chainSynced =
+      history.bestFullBlockOpt.map(_.id) == context.lastHeaderOpt.map(_.id)
+
+    if (chainUpdateDelayed || blockUpdateDelayed) {
+      val bestFullBlockOpt =
+        history.bestFullBlockOpt
+          .filter(_.id != lastMod.id)
+          .fold("")(fb => s"\n best full block: $fb")
+      val repairNeeded = ErgoHistory.repairIfNeeded(history)
+      ChainIsStuck(s"Chain not modified for $chainUpdateDelay ms, headers-height: $headersHeight, " +
+        s"block-height $blockHeight, chain synced: $chainSynced, repair needed: $repairNeeded, " +
+        s"last modifier applied: $lastMod $bestFullBlockOpt")
```

Comment:
I thought it would be useful to print out last persistent modifier applied to history, possibly bestFullBlock if they are not the same modifiers